### DISCUSSION TITLE
Can I control the filenames of the exported diagrams?

### DISCUSSION THREAD

**Discussion by chipbite** [structurizr/cli#149] [Q&A] [answered]
Hi! Thx for this very nice toolset and ideas. Experimenting. Running cli to get plantuml, to start. 
puml files are named `diagram type` + `counter`, e g it looks like this:

```
export
structurizr-Container-001-key.puml
structurizr-Container-001.puml
structurizr-SystemContext-001-key.puml
structurizr-SystemContext-001.puml
structurizr-SystemContext-002-key.puml
structurizr-SystemContext-002.puml
```

## **Question:** Is there a way to **control the filenames** of the _view exports_?

So that e g  `structurizr-SystemContext-001.puml` would be for example `KP FrontEnd - Context diagram.puml`, or maybe `structurizr-SystemContext-frontEnd.puml` (i e `structurizr-[]-[view-key].puml`) or similar...
(C4 dsl snippet for view below.)

### References
Found #103 which referred to https://github.com/structurizr/cli/blob/b19e8871876a3597bf021dc19c384978a148e208/src/main/java/com/structurizr/cli/export/ExportCommand.java#L168-L170

So it seems that the answer is no... correct? I am working out how to use integrate the diagrams in other places...

Thanks again!
BR! /marcus

```
        systemContext frontend {
            title "KP FrontEnd - Context diagram"
            include *
            autolayout lr
        }
```

**Comment by simonbrowndotje** [comment, marked as answer]
The filename is based upon the `key` of the view (e.g. https://docs.structurizr.com/dsl/language#systemcontext-view), which is automatically generated if not specified.

**Comment by chipbite** [comment]
Alright, thanks for the clarification!
